radix-engine/src/kernel/call_frame.rs

```diff
@@ -10,139 +10,221 @@ use radix_engine_interface::blueprints::resource::{
 };
 use radix_engine_interface::types::{LockHandle, NodeId, SubstateKey};
 
-use super::heap::Heap;
+use super::actor::{Actor, MethodActor};
+use super::heap::{Heap, HeapLockSubstateError, HeapRemoveModuleError, HeapRemoveNodeError};
 use super::kernel_api::LockInfo;
 
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct CallFrameUpdate {
-    pub nodes_to_move: Vec<NodeId>,
-    pub node_refs_to_copy: IndexSet<NodeId>,
+/// A message used for communication between call frames.
+///
+/// Note that it's just an intent, not checked/allowed by kernel yet.
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct Message {
```

Comment:
Can we change name, `Message` definitely does not give impression that this includes state movement?

Reply:
The reason I renamed `CallFrameUpdate` is to make it clear that it's an itent, not a result.

But, renaming suggestion welcome.